### PR TITLE
chore(Modal): Update tests to use `Trigger`, and open

### PR DESCRIPTION
**Pull request by Barsnes**

### CODE REVIEW

File: packages/react/src/components/Modal/Modal.test.tsx [hunk 1]
```diff
@@ -44,8 +30,67 @@ const render = async (props: Partial<ModalRootProps> = {}) => {
 };
 
 describe('Modal', () => {
+  beforeAll(() => {
+    HTMLDialogElement.prototype.show = vi.fn(function mock(
+      this: HTMLDialogElement,
+    ) {
+      this.open = true;
+    });
```
Comment (mimarz) [inline packages/react/src/components/Modal/Modal.test.tsx:38]: If you declare these mock functions outside we can also use them to assert that they have been called x times :)

**Review by mimarz — CHANGES_REQUESTED** [review]
Nice work! 

Added a suggestion on how to get better test coverage using the new mocks :D 

Do we need an afterAll to clean up or reset any mock code?

File: packages/react/src/components/Modal/Modal.test.tsx (hunk 1 above)
Comment (Barsnes) [reply packages/react/src/components/Modal/Modal.test.tsx:38]: Will move them to the setup file 😄

**Comment by Barsnes** [comment]
> Nice work!
> 
> Added a suggestion on how to get better test coverage using the new mocks :D
> 
> Do we need an afterAll to clean up or reset any mock code?

I don't think these mocks need to be reset, since it always recieves a new `this` for the dialog element

**Comment by mimarz** [comment]
I see you moved the mocks, but we aren't testing if they have been called though? Should we not also test that?

**Comment by Barsnes** [comment]
> I see you moved the mocks, but we aren't testing if they have been called though? Should we not also test that?

Do we need to test this?

**Comment by mimarz** [comment]
> > I see you moved the mocks, but we aren't testing if they have been called though? Should we not also test that?
> 
> Do we need to test this?

I guess not, but its easy to do now so 🤷  You decide, looks good either way.

**Review by mimarz — APPROVED** [review]

**Comment by Barsnes** [comment]
> > > I see you moved the mocks, but we aren't testing if they have been called though? Should we not also test that?
> > 
> > 
> > Do we need to test this?
> 
> I guess not, but its easy to do now so 🤷 You decide, looks good either way.

I went ahead and check them - does not hurt :D 
Let me know if everything still looks good

**Comment by mimarz** [comment]
> > > > I see you moved the mocks, but we aren't testing if they have been called though? Should we not also test that?
> > > 
> > > 
> > > Do we need to test this?
> > 
> > 
> > I guess not, but its easy to do now so 🤷 You decide, looks good either way.
> 
> I went ahead and checked them - does not hurt :D Let me know if everything still looks good

Looks nice!

**Pull request merged**